File: projects/dotnet-sdk/Controllers/VisitsController.cs
using Microsoft.AspNetCore.Mvc;
using FingerprintPro.ServerSdk.Api;
using FingerprintPro.ServerSdk.Client;
using FingerprintPro.ServerSdk.Model;
using dotnet_sdk.Models;

namespace dotnet_sdk.Controllers
{
    [ApiController]
    [Route("")]
    public class VisitsController : ControllerBase
    {
        [HttpGet("/getVisits")]
        public async Task<IActionResult> GetVisits(
            [FromQuery] string? apiKey,
            [FromQuery] string? region,
            [FromQuery] string? visitorId,
            [FromQuery] string? requestId,
            [FromQuery] string? linkedId,
            [FromQuery] int? limit,
            [FromQuery] string? paginationKey,
            [FromQuery] long? before
            )
        {
            try
            {
                var parsedRegion = Utils.GetRegion(region);
                var configuration = new Configuration(apiKey)
                {
                    Region = parsedRegion
                };
                var api = new FingerprintApi(configuration);

                var apiResponse = api.GetVisitsWithHttpInfo(visitorId, requestId, linkedId, limit, paginationKey, before);
                var eventResponse = apiResponse.Response;
                var rawResponse = await eventResponse.Content.ReadAsStringAsync();

                var response = new MusicianResponse<VisitorsGetResponse>(eventResponse.StatusCode, rawResponse, apiResponse.Data);
                return Ok(response);
            }
            catch (Exception e) {
                return Ok(Utils.ProcessException(e));
            }
        }

        [HttpGet("/deleteVisitorData")]
        public async Task<IActionResult> DeleteVisitorData(
            [FromQuery] string? apiKey,
            [FromQuery] string? region,
            [FromQuery] string? visitorId
            )
        {
            try
            {
                var parsedRegion = Utils.GetRegion(region);
                var configuration = new Configuration(apiKey)
                {
                    Region = parsedRegion
                };
                var api = new FingerprintApi(configuration);

                var apiResponse = api.DeleteVisitorDataWithHttpInfo(visitorId);
                var eventResponse = apiResponse.Response;
                var rawResponse = await eventResponse.Content.ReadAsStringAsync();

                var response = new MusicianResponse<object>(eventResponse.StatusCode, rawResponse, apiResponse.Data);
                return Ok(response);
            }
            catch (Exception e) {
                return Ok(Utils.ProcessException(e));
            }
        }
    }
}

File: projects/java-sdk/src/main/java/com/example/java_sdk/VisitsController.java
package com.example.java_sdk;

import com.fingerprint.api.FingerprintApi;
import com.fingerprint.model.EventsGetResponse;
import com.fingerprint.model.VisitorsGetResponse;
import com.fingerprint.sdk.ApiClient;
import com.fingerprint.sdk.ApiException;
import com.fingerprint.sdk.ApiResponse;
import com.fingerprint.sdk.Configuration;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

public class VisitsController {

    @GetMapping("/getVisits")
    public ResponseEntity<MusicianResponse> getVisits(
            @RequestParam(required = false, value = "") String apiKey,
            @RequestParam(required = false, value = "") String region,
            @RequestParam(required = false, value = "") String visitorId,
            @RequestParam(required = false) String requestId,
            @RequestParam(required = false) String linkedId,
            @RequestParam(required = false) int limit,
            @RequestParam(required = false) String paginationKey,
            @RequestParam(required = false) Long before
    ) {
        ApiClient client = Configuration.getDefaultApiClient(apiKey, Utils.getRegion(region));
        FingerprintApi api = new FingerprintApi(client);
        try {
            final ApiResponse<VisitorsGetResponse> apiResponse = api.getVisitsWithHttpInfo(visitorId, requestId, linkedId, limit, paginationKey, before);
            final VisitorsGetResponse visits = apiResponse.getData();
            final int code = apiResponse.getStatusCode();
            final MusicianResponse response = new MusicianResponse(code, visits, visits);
            return ResponseEntity.ok(response);
        } catch (ApiException e) {
            final MusicianResponse response = new MusicianResponse(e);
            return ResponseEntity.ok(response);
        }
    }

    @GetMapping("/deleteVisitorData")
    public ResponseEntity<MusicianResponse> deleteVisitorData(
            @RequestParam(required = false, value = "") String apiKey,
            @RequestParam(required = false, value = "") String region,
            @RequestParam(required = false, value = "") String visitorId
    ) {
        ApiClient client = Configuration.getDefaultApiClient(apiKey, Utils.getRegion(region));
        FingerprintApi api = new FingerprintApi(client);
        try {
            final ApiResponse<Void> apiResponse = api.deleteVisitorDataWithHttpInfo(visitorId);
            final int code = apiResponse.getStatusCode();
            final MusicianResponse response = new MusicianResponse(code, apiResponse.getData(), apiResponse.getData());
            return ResponseEntity.ok(response);
        } catch (ApiException e) {
            final MusicianResponse response = new MusicianResponse(e);
            return ResponseEntity.ok(response);
        }
    }
}
